app/mm-redux/actions/apps.ts

```diff
@@ -16,6 +16,30 @@ export function fetchAppBindings(userID: string, channelID: string): ActionFunc
         return dispatch(bindClientFunc({
             clientFunc: () => Client4.getAppsBindings(userID, channelID, teamID),
             onSuccess: AppsTypes.RECEIVED_APP_BINDINGS,
+            onFailure: AppsTypes.CLEAR_APP_BINDINGS,
         }));
     };
 }
+
+export function fetchThreadAppBindings(userID: string, channelID: string): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const channel = getChannel(getState(), channelID);
+        const teamID = channel?.team_id || '';
+
+        return dispatch(bindClientFunc({
+            clientFunc: async () => {
+                const bindings = await Client4.getAppsBindings(userID, channelID, teamID);
+                return {bindings, channelID};
+            },
+            onSuccess: AppsTypes.RECEIVED_THREAD_APP_BINDINGS,
+            onRequest: AppsTypes.CLEAR_THREAD_APP_BINDINGS,
```

Comment:
What's the reasoning of using `onFailure` for the above case, and using `onRequest` here?

Reply:
I wanted to make sure that if the call took long, the bindings were empty. Not sure if needed, and not sure if we should follow for the main bindings the same approach.

I know that for webapp we won't want something like this, because we would experience blinking in the channel header buttons (on request they disappear, on success they appear again). Since there are no visible bindings on mobile when changing channels, I thought this was a good approach.

If we ever have bindings visible, when switching channels or opening threads, we might want to rethink this approach.